app/src/main/java/ch/epfl/sdp/appart/database/FirestoreDatabaseService.java

```diff
@@ -301,22 +272,20 @@ public FirestoreDatabaseService() {
         CompletableFuture<Void> imagesResult = new CompletableFuture<>();
         CompletableFuture<Void> adResult = new CompletableFuture<>();
         CompletableFuture<Void> cardResult = new CompletableFuture<>();
-        DocumentReference newAdRef = db.collection(AdLayout.DIRECTORY).document();
-        DocumentReference cardRef = db.collection(CardLayout.DIRECTORY).document();
+        DocumentReference newAdRef = db.collection(FirebaseLayout.ADS_DIRECTORY).document();
+        DocumentReference cardRef = db.collection(FirebaseLayout.CARDS_DIRECTORY).document();
 
         // upload photos
         List<String> actualRefs = new ArrayList<>();
         List<CompletableFuture<Boolean>> imagesUploadResults = new ArrayList<>();
-        Log.d("URI", "size"+uriList.size());
+        Log.d("URI", "size" + uriList.size());
         for (int i = 0; i < uriList.size(); i++) {
-            String name = "photo" + ad.getAdvertiserId() + System.currentTimeMillis() + i + ".jpeg"; // getTypeFromUri(uriList.get(i));
+            String name = FirebaseLayout.PHOTO_NAME + i + ".jpeg"; // TODO modify to support other extensions
             actualRefs.add(name);
-            imagesUploadResults.add(putImage(uriList.get(i), name, "Ads"));
-            if(i == 0){
-                imagesUploadResults.add(putImage(uriList.get(0), name, "Cards"));
-            }
+            imagesUploadResults.add(putImage(uriList.get(i), name,
+                    FirebaseLayout.ADS_DIRECTORY + FirebaseLayout.SEPARATOR + newAdRef.getId()));
         }
-        // check whether any of the uploads failed
+        // check whether any of the uploads failed TODO check
```

Comment:
Left TODO

Reply:
whoops completely forgot about this, thank you!